### PR TITLE
CORE-31430 Fix issue where identity would only be requested on viewStart

### PR DESCRIPTION
**Pull request by Aaronius**
## Description
This solves the problem brought up in https://github.com/adobe/alloy/issues/69. That is, if the ECID is not available on the client and the website executes `event` commands that do NOT have `type: viewStart`, then the first request would be sent but no ECID requested. The response therefore did not have an ECID and subsequent requests would never be sent (they were stuck in the queue waiting for an ECID). This PR fixes the problem by requesting the ECID on the first event regardless of whether it is `type: viewStart`.

This PR also gives components greater control over whether a response is expected by exposing an `expectResponse` method on the event rather than basing the response expectation on the existence of a `query` object. We were going to need this anyway when gateway becomes less reliant on the `query` object for sending data back from the server.

## Related Issue
https://jira.corp.adobe.com/browse/CORE-31430
https://jira.corp.adobe.com/browse/CORE-31429

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] (all tests pass, but we need functional tests) I have added/removed/modified tests to cover my changes and all tests pass.
- [X] I have run the Sandbox successfully.

### CODE REVIEW

**Comment by jonsnyder** [comment]
It may be a good idea to log a debug message when the identity component is holding up a request.  I could see there being other reasons why this might happen.

File: src/components/Identity/index.js [hunk 1]
```diff
@@ -26,18 +26,19 @@ const createIdentity = ({ config, logger, cookie }) => {
...
-      return;
+  const onBeforeEvent = event => {
+    if (!ecid && !responseRequested) {
+      // TODO: Remove; We won't need to request id syncs explicitely.
+      // This is just for demo currently.
+      event.mergeQuery({
+        idSyncs: true
+      });
+      event.expectResponse();
+      responseRequested = true;
```
Comment (jonsnyder) [inline src/components/Identity/index.js]: Long term, we need to figure out a way for the identity component to know when the id request fails or times out.

**Comment by Aaronius** [comment]
Yeah, logging a message would be good. I'll do that right now.

File: src/components/Identity/index.js (hunk 1 above)
Comment (Aaronius) [reply src/components/Identity/index.js]: Yes...and what we want to do about it when it does fail or time out...

Comment (jfkhoury) [reply src/components/Identity/index.js]: We have this ticket that might cover that use case: https://jira.corp.adobe.com/browse/CORE-28205

**Pull request merged**